src/lowlevel/Random.cpp

```diff
@@ -52,9 +53,18 @@ int Random::get_number(unsigned int x) {
  * \param y the superior bound
  * \return a random integer number in [x, y[
  */
-int Random::get_number(unsigned int x, unsigned int y) {
-  return x + get_number(y - x);
-}
+int get_number(int x, int y) {
+
+  // Initialize the engine and the distribution
+  thread_local std::mt19937 engine(std::time(nullptr));
```

Comment:
Why do you initialize the engine every time a random number is asked, and not in initialize()?